### ISSUE TITLE
Blocks: Implement as classes

### ISSUE THREAD

**Issue by justinstenning** [chraft/c-raft#51]
Create abstract Block class, and descend each block type from this (e.g. a stone block, or button block, or pressure plate block and so on).

Each block as different properties such as:

Friction (soul sand, ice, other), a step sound, fall behaviour, logic (e.g. button/pressure plate) and so on.

Also some logic contained within WorldGen will need to move into appropriate block classes.

A factory class will provide an extension point for implementing custom block behaviour.

**Comment by taugit** [comment]
I'm working on it now. Preliminary code will be available soon.

**Comment by taugit** [comment]
I consider this task finished: blocks are implemented and BlockHelper works well. Current BaseBlock class is flexible enough  and all the upcoming improvements/fixes can be posted under a separate ticket.
